node/cn/filters/api.go

```diff
@@ -88,20 +92,30 @@ func NewPublicFilterAPI(backend Backend, lightMode bool) *PublicFilterAPI {
 // timeoutLoop runs every 5 minutes and deletes filters that have not been recently used.
 // Tt is started when the api is created.
 func (api *PublicFilterAPI) timeoutLoop() {
-	ticker := time.NewTicker(5 * time.Minute)
+	var toUninstall []*Subscription
+	ticker := time.NewTicker(api.timeout)
+	defer ticker.Stop()
 	for {
 		<-ticker.C
 		api.filtersMu.Lock()
```

Comment:
instead of `toUninstall = nil` at the loop tail,
```suggestion
		api.filtersMu.Lock()
		toUnsubscribe := []*Subscription{}
```

Reply:
Oh, nevermind. geth is using this code.